src/components/Editor/EditorInput/EditorInput.jsx

```diff
@@ -79,20 +79,24 @@ const EditorInput = () => {
   };
 
   const [numberOfComponents, setNumberOfComponents] = useState(
-    project.components.length,
+    project?.components?.length,
   );
-  let tabRefs = useRef(project.components.map(createRef));
+  let tabRefs = useRef(project?.components?.map(createRef));
 
   useEffect(() => {
+    if (!project?.components) return;
+
     setNumberOfComponents(project.components.length);
     Array(project.components.length)
       .fill()
       .forEach((_, i) => {
         tabRefs.current[i] = tabRefs.current[i] || React.createRef();
       });
-  }, [project.components.length]);
+  }, [project?.components]);
```

Comment:
🙌 this may solve the ongoing alert about undefined length!! 
![image](https://github.com/user-attachments/assets/287ab1ed-2113-44df-90e4-fdb244e092cd)